vmngclient/primitives/__init__.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING, Optional
+
+from packaging.specifiers import SpecifierSet  # type: ignore
+from packaging.version import Version  # type: ignore
+
+from vmngclient.exceptions import APIVersionException
+
+if TYPE_CHECKING:
+    from vmngclient.response import vManageResponse
+    from vmngclient.session import vManageSession
+
+
+BasePath = "/dataservice"
+logger = logging.getLogger(__name__)
+
+
+class APIPrimitiveBase:
+    def __init__(self, session: vManageSession):
+        self.session = session
+        self.basepath = BasePath
+
+    def request(self, method: str, urn: str, *args, **kwargs) -> vManageResponse:
+        return self.session.request(method, self.basepath + urn, *args, **kwargs)
+
+    def get(self, urn: str, *args, **kwargs) -> vManageResponse:
+        return self.request("GET", urn, *args, **kwargs)
+
+    def put(self, urn: str, *args, **kwargs) -> vManageResponse:
+        return self.request("PUT", urn, *args, **kwargs)
+
+    def post(self, urn: str, *args, **kwargs) -> vManageResponse:
+        return self.request("POST", urn, *args, **kwargs)
+
+    def delete(self, urn: str, *args, **kwargs) -> vManageResponse:
+        return self.request("DELETE", urn, *args, **kwargs)
+
+    @property
+    def version(self) -> Optional[Version]:
```

Comment:
I do not know what `Version` is, but maybe we should assign some `null` value inside `Version` class, like `UNKNOWN`

Reply:
Im afraid `packaging.version.Version` does not implement `UNKNOWN` version. What would be expected comparison with specifier set? For example does `UNKNOWN` belong to ">=20.6"?